lib/log/json_formatter.rb

```diff
@@ -0,0 +1,26 @@
+require 'json'
+
+module Log
+  class JSONFormatter
+
+    def call(severity, time, progname, msg)
+      log_item = {
+        app: progname,
+        host: host,
+        level: severity,
+      }
+      log_item[:message] = msg[:message] if msg[:message]
```

Comment:
I would be more flexible here. I think message could be anything as long as it fulfills a really simple interface.

You could maybe do something along the lines of

```ruby
log_item = {
  app: progname,
  host: host,
  level: severity,
}

case msg
when String 
  log_item[:message] = msg
when ->(msg) { msg.respond_to?(:to_hash) } 
  log_item[:message] = msg.to_hash
else
  log_item[:message] = msg.inspect
end
```